spec/forms/schools/add_participants/wizard_steps/email_step_spec.rb

```diff
@@ -42,6 +42,7 @@
     context "when registration is not yet open" do
       let(:automatically_assign_next_cohort) { true }
       let(:registration_open) { false }
+      before { allow(Cohort).to receive(:within_next_registration_period?).and_return(registration_open) }
```

Comment:
why this solution instead of using `mid_cohort: true` as above?
Isn't this solution implementation dependant?

Reply:
The convention in this test is to use stubs to create setup conditions, I have followed this.

Reply:
Given that we have a proper way (via that helper that travels to the right date) to simulate we are not opening registration yet, I would prefer to use it instead

Reply:
This is a unit test. The convention in the test is quite rightly to stub conditions for the set up of the test subject which I have followed with this change.
Introducing a date and time distortion via the `mid_cohort: true` or any other time travel mechanism is not only unnecessary but something we should only do in features.

Reply:
What is a real distorsion is stubbing a method that depends on the current time and leave the current time unchanged. 
That contradiction has been the cause of these and many other flacky tests during the life of this project.

Reply:
Anyway, not the right monent for these sort of discussions.